modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1234,7 +1236,7 @@ private CompletableFuture<?> createTableLocally(
 
         // TODO should be reworked in IGNITE-16763
         // We use the event notification future as the result so that dependent components can complete the schema updates.
-        return fireEvent(TableEvent.CREATE, new TableEventParameters(causalityToken, tableId));
+        return allOf(createPartsFut, fireEvent(TableEvent.CREATE, new TableEventParameters(causalityToken, tableId)));
```

Comment:
This fix will significantly slow down some tests, and the system in general when creating table/changing assignments. We need a better fix, which means adding TODO and describing what's happening and why the old code didn't work

Reply:
I don't really observe a slow down in tests, but ok.

Reply:
I mean that there seems to be no impact to teamcity build time

Reply:
There is, 20-40 seconds, as far as I remember